### PR TITLE
Allow platform default directives

### PR DESCRIPTION
**Pull request by wxtim**
This is a change with no associated Issue.

## The issue
Cylc 7 allowed the setting of default job runner (batch system) directives using [[test battery][batch systems][SYSTEM][directives]](https://cylc.github.io/cylc-doc/stable/html/appendices/site-user-config-ref.html#test-battery-batch-systems-system-directives).

Allow the setting of default directives in Cylc 8.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] Self documentation included in changes for `globalcfg.py`

### CODE REVIEW

**Review by hjoliver — APPROVED** [review]
Looks good, thanks @wxtim

File: CHANGES.md [hunk 1]
```diff
@@ -35,6 +35,9 @@ Fourth Release Candidate for Cylc 8 suitable for acceptance testing.
 
 ### Enhancements
 
+[#4896](https://github.com/cylc/cylc-flow/pull/4896) - Allow the setting of
+default job runner directives for plaforms.
```
Comment (MetRonnie) [inline CHANGES.md:39]:
```suggestion
default job runner directives for platforms.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -1313,6 +1313,17 @@
                       be appropriate if following the pattern
                       ``hosts = main, backup, failsafe``.
                 ''')
+            with Conf('directives', desc='''
+                Job runner (batch scheduler) directives.
+
+                Default directives for :cylc:conf:
+                `flow.cylc[runtime][<namespace>][directives]`
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1320]: Looks like the precedent for this kind of thing is
```suggestion
                :Defaults for: :cylc:conf:`flow.cylc[runtime][<namespace>][directives]`
```
(avoiding duplicating the description)

File: cylc/flow/task_job_mgr.py [hunk 3]
```diff
@@ -1299,7 +1320,9 @@ def get_job_conf(
                 itask.platform['job runner command template']
             ),
             'dependencies': itask.state.get_resolved_dependencies(),
-            'directives': rtconfig['directives'],
+            'directives': self.default_dict_merger(
+                itask.platform['directives'], rtconfig['directives']
+            ),
```
Comment (MetRonnie) [inline cylc/flow/task_job_mgr.py:1325]: Is it not possible to do?
```suggestion
            'directives': {
                **itask.platform['directives'], **rtconfig['directives']
            },
```

File: cylc/flow/cfgspec/globalcfg.py (hunk 2 above)
Comment (MetRonnie) [reply cylc/flow/cfgspec/globalcfg.py:1320]: Fair enough, in that case
```suggestion
                :Defaults for: :cylc:conf:`flow.cylc[runtime][<namespace>][directives]`
                
                Job runner (batch scheduler) directives.
```
`:Defaults for:` is some kind of sphinx directive by the look of it, possibly from cylc-sphinx-extensions

File: cylc/flow/task_job_mgr.py [hunk 4]
```diff
@@ -1299,7 +1299,9 @@ def get_job_conf(
                 itask.platform['job runner command template']
             ),
             'dependencies': itask.state.get_resolved_dependencies(),
-            'directives': rtconfig['directives'],
+            'directives': {
```
Comment (wxtim) [inline cylc/flow/task_job_mgr.py:1302]: @MetRonnie  - I knew there should be a 1-line for this!

**Comment by MetRonnie** [comment]
(Functional test failures due to Codecov upload failures)

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**